src/Core/src/RequestContext.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace AsyncAws\Core;
+
+use AsyncAws\Core\Exception\InvalidArgument;
+
+/**
+ * Contains contextual information alongside a request.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ *
+ * @internal
+ */
+class RequestContext
+{
+    public const AVAILABLE_OPTIONS = [
+        'operation' => true,
+        'expirationDate' => true,
+        'currentDate' => true,
+    ];
+
+    /**
+     * @var string|null
+     */
+    private $operation;
+
+    /**
+     * @var \DateTimeImmutable|null
```

Comment:
Should be `DateTimeInterface`. Or else it will not match the signature of `presign()`

Reply:
Good catch. I switched to DateTimeInterface 
=> it adds complexity to generate a default expiration date. Because `\DatetimeInterface::add` does not exists